xml/libvirt_guest_installation.xml

```diff
@@ -565,10 +567,145 @@ network=vnet_nated</screen>
+      </para>
+
+      <para>
+        When using <command>virt-install</command>, automatic firmware selection
+        is enabled by specifying the <emphasis>firmware=efi</emphasis> parameter
+        to the <emphasis>boot</emphasis> option, for example
+        <option>--boot firmware=efi</option>. The selection process can be
+        influenced by requesting the presence or absence of firmware features.
+        The following example illustrates automatic firmware selection with
+        &uefisecboot; disabled.
+      </para>
+
+<screen>&prompt.user;virt-install --connect qemu:///system --virt-type kvm \
+--name sle15sp5 --memory 1024 --disk size=10 --cdrom /dev/cdrom --graphics vnc \
+--boot firmware=efi,firmware.feature0.name=secure-boot,firmware.feature0.enabled=no \
+--os-variant sle15sp5</screen>
+
+      <note>
+        <para>
+          To ensure persistent &vmguest;s use the same firmware and variable
+          store throughout their lifetime, &libvirt; will record automatically
+          selected firmware in the &vmguest; XML configuration. Automatic
+          firmware selection is a one-time activity. Once firmware has been
+          selected, it will only change if the &vmguest; administrator
+          explicitly does so using the manual firmware selection method.
+        </para>
+      </note>
+
+      <para>
+        The <emphasis>loader</emphasis> and <emphasis>nvram</emphasis>
+        parameters are used for manual firmware selection. <emphasis>loader</emphasis>
+        is required, <emphasis>nvram</emphasis> defines an optional UEFI
+        variable store. The following example illustrates manual firmware
+        selection with secure boot enabled.
+      </para>
+
+<screen>&prompt.user;virt-install --connect qemu:///system --virt-type kvm \
+--name sle15sp5 --memory 1024 --disk size=10 --cdrom /dev/cdrom --graphics vnc \
+--boot loader=/usr/share/qemu/ovmf-x86_64-smm-code.bin,loader.readonly=yes,loader.type=pflash,loader.secure=yes,nvram.template=/usr/share/qemu/ovmf-x86_64-smm-vars.bin \
+--os-variant sle15sp5</screen>
+
+      <note>
+        <para>
+          &libvirt; cannot modify any characteristics of the UEFI firmwares. For
+          example it cannot disable &uefisecboot; in a firmware that has
+          &uefisecboot; enabled, even when specifying <emphasis>loader.secure=no</emphasis>.
+          &libvirt; will ensure the specified firmware is capable of satisfying
+          any specified features. For example it will reject configuration that
+          disables secure boot with <emphasis>loader.secure=no</emphasis>, but
+          specifies a firmware that has &uefisecboot; enabled.
+        </para>
+      </note>
+
+      <para>
+        The <package>qemu-ovmf-x86_64</package> package contains several UEFI
+        firmware images. For example the following subset supports SMM,
+        &uefisecboot; and have either Microsoft, openSUSE, or SUSE UEFI CA keys
```

Comment:
```suggestion
        firmware images. For example, the following subset supports SMM and
        &uefisecboot; and can have Microsoft, openSUSE, or SUSE UEFI CA keys
```

Reply:
I've applied some of this suggestion, and made a tiny adjustment myself.